pkg/volume/util/atomic_writer.go

```diff
@@ -300,27 +306,23 @@ func (w *AtomicWriter) shouldWriteFile(path string, content []byte) (bool, error
 	return (bytes.Compare(content, contentOnFs) != 0), nil
 }
 
-// pathsToRemove walks the user-visible portion of the target directory and
+// pathsToRemove walks the current version of the data directory and
 // determines which paths should be removed (if any) after the payload is
 // written to the target directory.
-func (w *AtomicWriter) pathsToRemove(payload map[string]FileProjection) (sets.String, error) {
+func (w *AtomicWriter) pathsToRemove(payload map[string]FileProjection, oldTsDir string) (sets.String, error) {
```

Comment:
Can we just use `dataDirName` instead of `oldTsDir` and avoid adding a param to the func?

Reply:
I don't think so because we need an absolute path for the `root` passed to `filepath.Walk()`. Also, because `filepath.Walk()` doesn't follow symlinks, we'd have to construct an absolute path out of `w.targetDir` and `dataDirName`, and then we'd need to follow that symlink to get the path. My thought is that it's better to follow the symlink once (to create `oldTsDir`) and pass that to anything that needs it.